instana/instrumentation/boto3_inst.py

```diff
@@ -35,6 +46,20 @@ def lambda_inject_context(payload, scope):
             logger.debug("non-fatal lambda_inject_context: ", exc_info=True)
 
 
+    @wrapt.patch_function_wrapper("botocore.hooks", "HierarchicalEmitter.emit_until_response")
+    def emit_until_response_with_instana(wrapped, instance, args, kwargs):
+        active_tracer = get_active_tracer()
+        
+        # If we're not tracing or the event emitted is not before-call, just return;
+        if active_tracer is None or args[0].split(".")[0] != "before-call":
+            return wrapped(*args, **kwargs)
+        
+        span = active_tracer.active_span
+        if "custom_request_headers" in kwargs["context"]:
```

Comment:
1. As it turned out there is no official way for setting custom request headers on boto3.
See https://github.com/boto/boto3/issues/2889 & https://github.com/boto/botocore/issues/2050.

2. Any header, that could be used for an HTTP connection (including the built-in ones or any custom ones) are in the `kwargs['params']['headers']` dictionary in this code scope. That is what we could extract from. Anything else is non relevant from tracing perspective. But the default request headers are rather minimal, so minimal value for the customer.

3. Apparently the most accepted "workaround" for setting custom request headers is this one:
https://stackoverflow.com/questions/58828800/adding-custom-headers-to-all-boto3-requests
but there could be variations.

I am fine with either:
A, documenting the resons, and completely omitting the extraction from request headers.
B, only extract from default headers in  `kwargs['params']['headers']`